### PR TITLE
[MB-1433] Create queue when publishing

### PR DESCRIPTION
**Pull request by sdkottegoda**
Addresses the issue reported at https://wso2.org/jira/browse/MB-1433

Added queues and bindings when creating a publisher.

### CODE REVIEW

**Comment by hemikak** [comment]
Please confirm with this too : https://wso2.org/jira/browse/MB-1185

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java [hunk 1]
```diff
@@ -2971,10 +2981,25 @@ private void registerConsumer(C consumer, boolean nowait) throws AMQException //
...
+                declareExchange(amqDestination, protocolHandler, nowait);
+            }
+            if (DECLARE_QUEUES || amqDestination.isNameRequired()) {
+                declareQueue(amqDestination, protocolHandler, false, nowait);
+            }
+            bindQueue(amqDestination.getAMQQueueName(), amqDestination.getRoutingKey(), FieldTableFactory
+                    .newFieldTable(), amqDestination.getExchangeName(), amqDestination, nowait);
+            _producers.put(new Long(producerId), producer);
+        } catch (JMSException e) {
+            throw e;
```
Comment (sajinidesilva) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java]: Why are we catching the exception just to throw it. If no processing is required no point of catching.

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java [hunk 2]
```diff
@@ -2971,10 +2981,25 @@ private void registerConsumer(C consumer, boolean nowait) throws AMQException //
...
-    
-    private void registerProducer(long producerId, MessageProducer producer)
-    {
-        _producers.put(new Long(producerId), producer);
+
+    private void registerProducer(long producerId, P producer) throws AMQException, JMSException {
+        try {
+            AMQDestination amqDestination = producer.getDestination();
+            AMQProtocolHandler protocolHandler = getProtocolHandler();
+            boolean nowait = false;
```
Comment (sajinidesilva) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java]: Variable name convention is wrong. Should be noWait.

File: modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java [hunk 3]
```diff
@@ -2971,10 +2981,25 @@ private void registerConsumer(C consumer, boolean nowait) throws AMQException //
     public abstract void handleAddressBasedDestination(AMQDestination dest, 
                                                        boolean isConsumer,
                                                        boolean noWait) throws AMQException;
-    
-    private void registerProducer(long producerId, MessageProducer producer)
-    {
-        _producers.put(new Long(producerId), producer);
+
+    private void registerProducer(long producerId, P producer) throws AMQException, JMSException {
```
Comment (sajinidesilva) [inline modules/andes-core/client/src/main/java/org/wso2/andes/client/AMQSession.java]: Shall we add a comment to this method?

**Pull request merged**